engine/src/cython/io.cpp

```diff
@@ -47,41 +66,65 @@ TableSchema parseSchema(std::vector<std::string> files,
 		parser = std::make_shared<ral::io::json_parser>(args_map);
 	} else if(fileType == ral::io::DataType::CSV) {
 		parser = std::make_shared<ral::io::csv_parser>(args_map);
-	}
+	} else if(fileType == ral::io::DataType::MYSQL) {
+#ifdef MYSQL_SUPPORT
+		parser = std::make_shared<ral::io::mysql_parser>();
+    auto sql = ral::io::getSqlInfo(args_map);
+    provider = std::make_shared<ral::io::mysql_data_provider>(sql);
+#endif
+    isSqlProvider = true;
+  } else if(fileType == ral::io::DataType::SQLITE) {
+#ifdef SQLITE_SUPPORT
+//		parser = std::make_shared<ral::io::sqlite_parser>();
+//    auto sql = ral::io::getSqlInfo(args_map);
+//    provider = std::make_shared<ral::io::sqlite_data_provider>(sql.schema,
+//                                                               sql.table,
+//                                                               sql.table_filter,                                                          sql.table_batch_size);
+//    isSqlProvider = true;
+#endif
+  }
+
+  if (!isSqlProvider) {
+      provider = std::make_shared<ral::io::uri_data_provider>(uris, ignore_missing_paths);
+  }
 
-	std::vector<Uri> uris;
-	for(auto file_path : files) {
-		uris.push_back(Uri{file_path});
-	}
-	auto provider = std::make_shared<ral::io::uri_data_provider>(uris, ignore_missing_paths);
 	auto loader = std::make_shared<ral::io::data_loader>(parser, provider);
 
 	ral::io::Schema schema;
 
 	try {
-
 		bool got_schema = false;
-		while (!got_schema && provider->has_next()){
-			ral::io::data_handle handle = provider->get_next();
-			if (handle.file_handle != nullptr){
-				parser->parse_schema(handle.file_handle, schema);
-				if (schema.get_num_columns() > 0){
-					got_schema = true;
-					schema.add_file(handle.uri.toString(true));
-				}
-			}
+    if (isSqlProvider) {
+        ral::io::data_handle handle = provider->get_next(false);
+        parser->parse_schema(handle, schema);
+        if (schema.get_num_columns() > 0){
+          got_schema = true;
+        }
+    } else {
+      while (!got_schema && provider->has_next()){
+        ral::io::data_handle handle = provider->get_next();
```

Comment:
What if we make `data_handle` an interface instead of a struct where we are shoving more mutually exclusive stuff into it?

If we did that, we could replace `handle.file_handle != nullptr` with `handle.is_valid()` which would apply to any type of data_handle and make a lot of other things cleaner

Reply:
Yes the idea is to have a nice interface here, however I think that is not part of the scope of this PR. There are many ways create a better `data_handle` concept and to really be data_source agnostic, but we lack the concept of data_source per se in our engine. I would propose to take this issue later, not for this PR cc @felipeblazing @gcca